Tests/LibAudio/TestWav.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024, Lee Hanken <github-12-2017-ds8@leehanken.uk>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ByteString.h>
+#include <AK/LexicalPath.h>
+#include <AK/ScopeGuard.h>
+#include <LibAudio/WavLoader.h>
+#include <LibAudio/WavWriter.h>
+#include <LibCore/Directory.h>
+#include <LibCore/File.h>
+#include <LibFileSystem/FileSystem.h>
+#include <LibFileSystem/TempFile.h>
+#include <LibTest/TestCase.h>
+
+static void compare_files(ByteString const in_path, ByteString const out_path)
+{
+    Array<u8, 4096> buffer1;
+    Array<u8, 4096> buffer2;
+
+    auto original_file = MUST(Core::File::open(in_path, Core::File::OpenMode::Read));
+    auto copied_file = MUST(Core::File::open(out_path, Core::File::OpenMode::Read));
+
+    while (!original_file->is_eof() && !copied_file->is_eof()) {
+        auto original_bytes = TRY_OR_FAIL(original_file->read_some(buffer1));
+        auto copied_bytes = TRY_OR_FAIL(copied_file->read_some(buffer2));
+
+        EXPECT_EQ(original_bytes, copied_bytes);
+    }
+}
+
+static void run_test(StringView const file_name, int const num_samples, int channels, u32 const rate)
+{
+
+    const ByteString format = "RIFF WAVE (.wav)";
+    int const bits = 16;
+
+    auto out_path = ByteString(TRY_OR_FAIL(FileSystem::TempFile::create_temp_file())->path());
+
+    ScopeGuard remove_temporary_file_guard = [&] {
+        if (FileSystem::exists(out_path)) {
+            MUST(FileSystem::remove(out_path, FileSystem::RecursionMode::Disallowed));
+        }
+    };
```

Comment:
I don't understand why that is needed.

Reply:
Automatic temp file removal is not something that happens while a system is running, usually.